### ISSUE TITLE
Deployment error 

### ISSUE THREAD

**Issue by kerwinxu** [cotes2020/jekyll-theme-chirpy#820]
## Checklist

Run bundle exec jekyll b -d "_site/"
Configuration file: /home/runner/work/kerwinxu.github.io/kerwinxu.github.io/_config.yml
 Theme Config file: /home/runner/work/kerwinxu.github.io/kerwinxu.github.io/_config.yml
            Source: /home/runner/work/kerwinxu.github.io/kerwinxu.github.io
       Destination: /home/runner/work/kerwinxu.github.io/kerwinxu.github.io/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
  Liquid Exception: undefined method `tainted?' for "条形码管理专家":String in /home/runner/work/kerwinxu.github.io/kerwinxu.github.io/_layouts/post.html
                    ------------------------------------------------
      Jekyll 4.3.1   Please append `--trace` to the `build` command 
                     for any additional information or backtrace. 
                    ------------------------------------------------
/home/runner/work/kerwinxu.github.io/kerwinxu.github.io/vendor/bundle/ruby/3.2.0/gems/liquid-4.0.3/lib/liquid/variable.rb:[12](https://github.com/kerwinxu/kerwinxu.github.io/actions/runs/3788075813/jobs/6440484559#step:5:13)4:in `taint_check': undefined method `tainted?' for "条形码管理专家":String (NoMethodError)

      return unless obj.tainted?
                       ^^^^^^^^^

file : 2021-10-28-条形码管理专家.md
---
layout: post
title:  "条形码管理专家"
date:   2021-10-28 14:38:52 +0800
categories: 软件
project: true
excerpt: "支持自定义模板和批量打印"
image: "/assets/image/barcodemanager/img1.png"
lang: zh
tag:
- 条形码打印
- 批量打印
---

# 简介

方便的打印条形码的，支持自定义模板，支持批量打印。  
   - 模板编辑编辑模板。
   - 支持excel导入数据。
     - 支持冗余打印
     
# 下载

[最新下载](/assets/software/BarcodeManager.zip)

# 简单步骤
### 页面设置
打开软件后首先应该"页面设置"，也就是设置纸张  

![页面设置](/assets/image/barcodemanager/1.png)
不同的打印机有不同的纸张尺寸，请选择相关的打印机和相关的纸张尺寸，这里清注意，比如有一些A4纸大小的条形码纸张，还分成很多行列的，在这里，需要选择A4纸大小，然后行列数，请在“布局”中设置，对打印机而言，这个纸张尺寸就是A4大小，但对于我们想分隔成很多行列的，我们可以自己在这里设置  
![设置纸张](/assets/image/barcodemanager/2.png)  
### 编辑
这里可以做图形的编辑，重点的是条形码,右边可以设置相关的属性。  
![编辑](/assets/image/barcodemanager/3.png)  
条形码支持从excel导入输入，excel文件的格式如下，其中第一行为这一列的标题

|款号 |   品名  |  数量  |  条形码 |
| :--- | :---    | :---    | :---     |
|1 |   男装T恤  |  342 |   123456789012|
|2 |   女装T恤   | 42342 |   123456789013|
|3 |   男装休闲长裤  |  4242 |   123456789014 |
|4 |   女装休闲长裤  |  42  |  123456789015|
|5  |  男装牛仔长裤   | 4242 |   123456789016|
|6  |  女装牛仔长裤   | 42  |  123456789017 |
|7 |   男装长袖衬衫  |  452 |   123456789018|
|8 |   女装长袖衬衫  |  5432 |   123456789019|

这个标题就是变量，打印的时候，会按照excel的行读取，然后按照列名填充相关的内容。    

![变量](/assets/image/barcodemanager/4.png) 

## 条形码管理专家

这个重点是两个  
   - 模板 ： 可以导入用模板编辑的模板。
   - 数据 ： 通过excel导入

![条形码管理专家](/assets/image/barcodemanager/5.png) 

- [ ] I have read the [tutorials](https://cotes2020.github.io/chirpy-demo/categories/tutorial/) and know the correct effect of the functional design.
- [ ] There are no similar reports on [existing issues](https://github.com/cotes2020/jekyll-theme-chirpy/issues?q=is%3Aissue) (including closed ones).
- [ ] I found the bug on the latest code of the `master` branch.

### Environment

| Command                           | Version |
|-----------------------------------|---------|
| `ruby -v`                         |         |
| `gem -v`                          |         |
| `bundle -v`                       |         |
| `bundle exec jekyll -v`           |         |
| `bundle info jekyll-theme-chirpy` |         |

**Comment by xaviermignot** [comment]
Hello,

I'm having the same issue, deployment fails from GH Actions, it was working 2 days ago. Here is the error message:
```sh
  Liquid Exception: undefined method `tainted?' for "Hello world !":String in /home/runner/work/xaviermignot.github.io/xaviermignot.github.io/vendor/bundle/ruby/3.2.0/gems/jekyll-theme-chirpy-5.3.2/_layouts/post.html
                    ------------------------------------------------
      Jekyll 4.3.1   Please append `--trace` to the `build` command 
                     for any additional information or backtrace. 
                    ------------------------------------------------
```
I have close to no experience in Ruby but trying to figure out what is going on here.
I have compared my latest success run with the failing ones, I'm suspecting the recent [updates](https://github.com/ruby/setup-ruby/releases) of the `setup-ruby` action to cause the error. There has been 3 releases over the last two days, updating ruby from 3.1 to 3.2, bundler from 2.3 to 2.4, etc.

As a workaround I have enforced the version of the `setup-ruby` action in my repo by setting a tag before those releases, and it worked. See the [commit](https://github.com/xaviermignot/xaviermignot.github.io/commit/4b57358b0705cc4f328944fcba74f83e8253a3c4) in my repo if you want to do the same on your side and let your site deploy again.

This should work while someone more experienced with Ruby finds out why the latest releases have broken the workflow :)

**Comment by kerwinxu** [comment]
> 

Thank you for your reply . Could you tell me where setup-ruby i can edit ?

**Comment by NichtsHsu** [comment]
It was fixed at 09e4ad245b5f1101eacbc55b89e8e8dee05df746, please update your repo with new commits.

**Comment by xaviermignot** [comment]
Works great, thanks :)
